pkg/kubelet/network/kubenet/kubenet_linux.go

```diff
@@ -107,7 +108,16 @@ type kubenetNetworkPlugin struct {
 }
 
 func NewPlugin(networkPluginDir string) network.NetworkPlugin {
-	protocol := utiliptables.ProtocolIpv4
+	var protocol utiliptables.Protocol
+	var addr net.IP
+	addr, _ = utilnet.ChooseBindAddress(addr)
+	if addr.To4() != nil {
+		protocol = utiliptables.ProtocolIpv4
+	}
+	if addr.To16() != nil {
```

Comment:
I think you need to place this in an else clause, as To16() will return non-nil, for IPv4 addresses.